typescript-loops/main.ts

```diff
@@ -0,0 +1,63 @@
+/* exported getNumbersToTen,
+            getEvenNumbersToTwenty,
+            repeatWord,
+            logEachCharacter,
+            doubleAll,
+            getKeys,
+            getValues
+ */
+
+function getNumbersToTen(): number[] {
+  const numbers = [];
+  let currentNumber = 1;
+  while (currentNumber <= 10) {
+    numbers.push(currentNumber);
+    currentNumber++;
+  }
+  return numbers;
+}
+
+function getEvenNumbersToTwenty(): number[] {
+  const evenNumbers = [];
+  let currentNumber = 2;
+  while (currentNumber <= 20) {
+    evenNumbers.push(currentNumber);
+    currentNumber += 2;
+  }
+  return evenNumbers;
+}
+function repeatWord(word: string, times: number): string {
+  let count = 1;
+  let repeated = '';
+  while (count <= times) {
+    repeated += word;
+    count++;
+  }
+  return repeated;
+}
+function logEachCharacter(str: string): void {
+  for (let i = 0; i < str.length; i++) {
+    console.log(str[i]);
+  }
+}
+function doubleAll(numbers: []): number[] {
+  const doubled = [];
+  for (let i = 0; i < numbers.length; i++) {
+    doubled.push(numbers[i] * 2);
+  }
+  return doubled;
+}
+function getKeys(obj: Record<string, unknown>): unknown[] {
+  const key = [];
+  for (const prop in obj) {
+    key.push(prop);
+  }
+  return key;
+}
```

Comment:
If your `keys` are type `string` as you noted in `Record<string, unknown>`, does that mean we are returning an `unknown[]` or a `string[]`? 🤔

Reply:
It would be type string[] because I am returning the keys and I declared it as type string. Also wouldn't my key variable have to be a let variable instead of const, and why did it still let me change the const variable?